index.bs

```diff
@@ -3181,6 +3251,10 @@ To <dfn>process an attribution source</dfn> given an [=attribution source=] |sou
 1. Set |source|'s [=attribution source/number of event-level reports=] to 0 if
     |source|'s [=attribution source/randomized response=] is null,
     [=attribution source/randomized response=]'s [=list/size=] otherwise.
+1. Let |states| be the [=obtain a set of possible trigger states|number of possible trigger states=] obtained with |randomizedResponseConfig|.
+1. If |sourceType| is "<code>[=source type/event=]</code>" and |states| is greater than |source|'s [=attribution source/max event states=]:
```

Comment:
Shouldn't we only be performing this check if the scope limit isn't null? Otherwise, we're going to start rejecting registrations with > 3 states that aren't using scopes at all.

Reply:
https://github.com/WICG/attribution-reporting-api/pull/1386

Reply:
With this change, allowing max event states to 3 without attribution scopes limit doesn't seem to make sense any more. When there's no attribution scopes, we probably should disallow max event states to be set (stricter), or treat it as unknown fields as it's irrelevant (less strict). And max event states should be default to null when there's no attribution scopes, or 3 when there is.

As a side note, it seems to be much easier to wrap the three fields into a dictionary for clearness, but I assume it's too late to make such a change.

Reply:
If we think it makes sense to put the 3 options into a dictionary, I think we can do so.

Reply:
> Shouldn't we only be performing this check if the scope limit isn't null? Otherwise, we're going to start rejecting registrations with > 3 states that aren't using scopes at all.

As Nan pointed out, it might not be the best option to wrap the three fields into a dictionary as it will significantly push the Rb release. I see the benefit of that, but am still on the fence whether adding another layer makes it cleaner.

Thanks for catching and fixing this @apasel422

Reply:
> as it will significantly push the Rb release

Should that really be a concern when we're trying to specify something that will be easy to understand by our users?

https://www.w3.org/TR/design-principles/#priority-of-constituencies

Reply:
this is resolved in https://github.com/WICG/attribution-reporting-api/pull/1391

Reply:
this is resolved in https://github.com/WICG/attribution-reporting-api/pull/1391